src/TrackGenerator.cpp

```diff
@@ -677,6 +684,7 @@ void TrackGenerator::initializeTracks() {
   double* phi_eff = new double[_num_azim];
 
   double x1, x2;
+  double z = _geometry->getZLevel();
```

Comment:
It could be transferred and I agree that intuitively, it seems more fitting to have it in the `TrackGenerator`. However, doing this would then require that the plotter either use some other default zcoord (e.g. 0.0) for plotting the materials, cells, etc. I'll make it a `TrackGenerator` attribute and have the default zcoords in the plotter be 0.0.